core/Makefile

```diff
@@ -108,8 +108,8 @@ endif
 # these headers.  Should fix the warnings.  Using -isystem also disables
 # -MMD dependency recording (should we use -MD?).
 COREDIR := $(abspath .)
-CXXARCHFLAGS ?= -march=native
-CXXFLAGS += -std=c++17 -g3 -ggdb3 $(CXXARCHFLAGS) \
+CPU ?= native
+CXXFLAGS += -std=c++17 -g3 -ggdb3 -march=$(CPU) \
```

Comment:
I'm fine with this change, I'll just note that the old way allowed one to not specify `-march=...` at all by setting `CXXARCHFLAGS` to something weird. 😺 (This way doesn't do that but seems clearer / cleaner to me)